EarlGrey/Action/GREYActions.m

```diff
@@ -136,12 +136,10 @@ @implementation GREYActions
 }
 
 + (id<GREYAction>)actionForTurnSwitchOn:(BOOL)on {
-  id<GREYMatcher> constraints = grey_allOf(grey_not(grey_systemAlertViewShown()),
```

Comment:
Let's remove the matcher from action constraints and have a universal check in GREYElementInteraction, just like we have a universal check for element not being nil before calling GREYAction perform:error:. We can keep the matcher in GREYMatchers, but consider it deprecated, in order to not break backwards compatibility.

A matcher is not the right place for this check, because it is not element-specific, it is for system state.